cmd/subnetcmd/publish.go

```diff
@@ -395,43 +399,45 @@ func getVMInfo(sc *models.Sidecar) (*types.VM, error) {
 			return nil, errors.New("canceled by user")
 		}
 
+		url, err = app.Prompt.CaptureStringAllowEmpty("Tell us the URL to download the source. Needs to be a fixed version, not `latest`.")
+		if err != nil {
+			return nil, err
+		}
+
+		sha, err = app.Prompt.CaptureStringAllowEmpty("For integrity checks, provide the sha256 commit for the used version")
+		if err != nil {
+			return nil, err
+		}
+		strVer, err := app.Prompt.CaptureVersion("This is the last question! What is the version being used? Use semantic versioning (v1.2.3)")
```

Comment:
can we get version from url?

Reply:
As this is a custom VM, I am not sure we can safely derive the version from the url - the URL structure will be designed by the user, not us.